llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -271,6 +271,71 @@ static void collectKernelModuleMap(
   }
 }
 
+// Go through function call graph searching for assert call.
+static bool hasAssertInFunctionCallGraph(llvm::Function *Func) {
+  // Map holds the info about assertions in already examined functions:
+  // true  - if there is an assertion in underlying functions,
+  // false - if there are definetely no assertions in underlying functions.
+  static std::map<llvm::Function *, bool> hasAssertionInCallGraphMap;
+  std::vector<llvm::Function *> FuncCallStack;
+
+  std::vector<llvm::Function *> Workstack;
+  Workstack.push_back(Func);
+
+  while (!Workstack.empty()) {
+    Function *F = Workstack.back();
+    Workstack.pop_back();
+    if (F != Func)
+      FuncCallStack.push_back(F);
+
+    bool IsLeaf = true;
+    for (auto &I : instructions(F)) {
+      if (!isa<CallBase>(&I))
+        continue;
+
+      Function *CF = cast<CallBase>(&I)->getCalledFunction();
+      if (!CF)
+        continue;
+
+      // Return if we've already discovered if there are asserts in the
+      // function call graph.
+      if (hasAssertionInCallGraphMap.count(CF)) {
+        // If we know, that this function does not contain assert, we still
+        // should investigate another instructions in the function.
+        if (!hasAssertionInCallGraphMap[CF])
```

Comment:
map::find method may be used to search CF in hasAssertionInCallGraphMap once here.